### PR TITLE
Added some variants of Player.AddItem() method

### PR DESCRIPTION
**Pull request by NaoUnderscore**
Those're my extensions that I'm currently using to avoid filling the code with for / foreach loops.
May they be useful?

### CODE REVIEW

**Comment by NaoUnderscore** [comment]
fuck autosave - "Create Player.cs"

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -1313,12 +1313,66 @@ public void Broadcast(ushort duration, string message, global::Broadcast.Broadca
...
+        /// <summary>
+        /// Add the list of items of the specified type with default durability(ammo/charge) and no mods to the player's inventory.
+        /// </summary>
+        /// <param name="items">The list of items to be added.</param>
+        public void AddItem(List<ItemType> items)
+        {
+            if (items.Count > 0)
+            {
+                foreach (ItemType itemType in items)
+                    Inventory.AddNewItem(itemType);
```
Comment (joker-119) [inline Exiled.API/Features/Player.cs:1339]:
```suggestion
                for (int i = 0; i < items.Count; i++)
                    AddItem(items[i]);
```

There's no reason for us to use a foreach() with variable assignment here.

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -1313,12 +1313,66 @@ public void Broadcast(ushort duration, string message, global::Broadcast.Broadca
...
+        /// Add the amount of items to the player's inventory.
+        /// </summary>
+        /// <param name="item">The item to be added.</param>
+        /// <param name="amount">The amount of items to be added.</param>
+        public void AddItem(Inventory.SyncItemInfo item, int amount)
+        {
+            if (amount > 0)
+            {
+                for (int i = 0; i < amount; i++)
+                    Inventory.AddNewItem(item.id, item.durability, item.modSight, item.modBarrel, item.modOther);
```
Comment (joker-119) [inline Exiled.API/Features/Player.cs:1359]:
```suggestion
                    AddItem(item);
```

File: Exiled.API/Features/Player.cs [hunk 3]
```diff
@@ -1313,12 +1313,66 @@ public void Broadcast(ushort duration, string message, global::Broadcast.Broadca
...
+        /// <summary>
+        /// Add the list of items to the player's inventory.
+        /// </summary>
+        /// <param name="items">The list of items to be added.</param>
+        public void AddItem(List<Inventory.SyncItemInfo> items)
+        {
+            if (items.Count > 0)
+            {
+                foreach (Inventory.SyncItemInfo item in items)
+                    Inventory.AddNewItem(item.id, item.durability, item.modSight, item.modBarrel, item.modOther);
```
Comment (joker-119) [inline Exiled.API/Features/Player.cs:1372]:
```suggestion
                for (int i = 0; i < items.Count; i++)
                    AddItem(items[i]);
```

File: Exiled.API/Features/Player.cs [hunk 4]
```diff
@@ -1313,12 +1313,66 @@ public void Broadcast(ushort duration, string message, global::Broadcast.Broadca
...
+        /// Add the amount of items of the specified type with default durability(ammo/charge) and no mods to the player's inventory.
+        /// </summary>
+        /// <param name="itemType">The item to be added.</param>
+        /// <param name="amount">The amount of items to be added.</param>
+        public void AddItem(ItemType itemType, int amount)
+        {
+            if (amount > 0)
+            {
+                for (int i = 0; i < amount; i++)
+                    Inventory.AddNewItem(itemType);
```
Comment (joker-119) [inline Exiled.API/Features/Player.cs:1326]:
```suggestion
                    AddItem(itemType);
```

**Review by joker-119 — CHANGES_REQUESTED** [review]

**Review by iopietro — APPROVED** [review]

**Review by joker-119 — APPROVED** [review]

**Pull request merged**